cirq/google/optimizers/convert_to_sycamore_gates.py

```diff
@@ -51,12 +51,18 @@ def __init__(self,
         """
         Args:
             tabulation: If set, a tabulation for the Sycamore gate to use for
-                decomposing gates.
+                decomposing Matrix gates. If unset, an analytic calculation is
+                used for Matrix gates. To get a GateTabulation, provide its
+                constructor a base gate (in this case, usually cirq.google.SYC)
```

Comment:
oh interesting. So the tabulation is done in the constructor of `GateTabulation`? It might be slightly nicer to have a function that does the tabulation (i.e. the hard work) and returns a filled-in `GateTabulation` object. I personally don't expect a constructor to do work. This is probably related to my dislike for over-oop (in python; I know it's different in other languages). 

However, this is fine at present

Reply:
Actually, the code behaves as you suggest. The `GateTabulation` object just stores the result of the tabulation and uses it to compile gates. The actual construction is done with `gate_product_tabulation`, which Kevin moved [here](https://github.com/quantumlib/Cirq/blob/04e4fbcce767ac342b10f56cf185ac9dba4a181b/cirq/google/optimizers/two_qubit_gates/gate_compilation.py).

Reply:
Woops, updated the comment. Thanks for catching!

Reply:
ah good. yeah that function name is what I wanted documented :)